dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/DhisOidcProviderRepository.java

```diff
@@ -75,14 +77,30 @@ public void addRegistration( DhisOidcClientRegistration registration )
             return;
         }
 
-        registrationHashMap.put( registration.getClientRegistration().getRegistrationId(), registration );
+        registrationHashMap.putIfAbsent( registration.getClientRegistration().getRegistrationId(), registration );
+    }
+
+    public void removeRegistration( DhisOidcClientRegistration registration )
+    {
+        registrationHashMap.remove( registration.getClientRegistration().getRegistrationId() );
+    }
+
+    public void removeRegistration( String registrationId )
+    {
+        DhisOidcClientRegistration remove = registrationHashMap.remove( registrationId );
+
+        log.info( "removed:" + remove );
```

Comment:
Use {} argument annotation, and elaborate a bit more in the log message to make it more useful.

Reply:
This is debug code, removed it now.